imgparse/parser.py

```diff
@@ -0,0 +1,553 @@
+                self.image_path, self.s3_role, self._raw_data
+            )
+        return self._xmp_data
+
+    @property
+    def xmp_tags(self) -> xmp_tags.XMPTags:
+        """Get the xmp tag names associated with the image's sensor."""
+        if self.make() == "Sentera":
+            return xmp_tags.SenteraTags()
+        elif self.make() == "DJI" or self.make() == "Hasselblad":
+            return xmp_tags.DJITags()
+        elif self.make() == "MicaSense":
+            return xmp_tags.MicaSenseTags()
+        elif self.make() == "Parrot":
+            return xmp_tags.ParrotTags()
+        else:
+            return xmp_tags.XMPTags()
+
+    def make(self) -> str:
+        """Get the make and model of the sensor that took the image."""
+        try:
+            return str(self.exif_data["Image Make"].values)
+        except KeyError:
+            raise ParsingError(
+                "Couldn't parse the make and model. Sensor might not be supported"
+            )
+
+    def model(self) -> str:
+        """Get the make and model of the sensor that took the image."""
+        try:
+            return str(self.exif_data["Image Model"].values)
+        except KeyError:
+            raise ParsingError(
+                "Couldn't parse the make and model. Sensor might not be supported"
+            )
+
+    def firmware_version(self) -> Version:
+        """
+        Get the firmware version of the sensor.
+
+        Expects camera firmware version to be in semver format (i.e. MAJOR.MINOR.PATCH), with an optional 'v'
+        at the beginning.
+        """
+        try:
+            version_match = re.search(
+                "[0-9]+.[0-9]+.[0-9]+", self.exif_data["Image Software"].values
+            )
+            if not version_match:
+                raise KeyError()
+            major, minor, patch = version_match.group(0).split(".")
+        except (KeyError, ValueError):
+            raise ParsingError(
+                "Couldn't parse sensor version. Sensor might not be supported"
+            )
+
+        return Version(int(major), int(minor), int(patch))
+
+    def timestamp(self, solar_time: bool = False) -> datetime:
+        """
+        Get the time stamp of an image and parse it into a `datetime` object.
+
+        For Sentera sensors, the parsed datetime will be in utc. For DJI sensors, it will
+        be the local time on the sensor. Set `solar_time=True` to convert Sentera timestamps
+        to the approximate solar time based on the sensor's longitude.
+        """
+        try:
+            timestamp = datetime.strptime(
+                self.exif_data["EXIF DateTimeOriginal"].values, "%Y:%m:%d %H:%M:%S"
+            )
+
+            if solar_time and self.make() == "Sentera":
+                _, lon = self.location()
+                # longitude / 15 gives solar time offset
+                offset_seconds = int((lon / 15.0) * 3600)
+                timestamp += timedelta(seconds=offset_seconds)
```

Comment:
This is really nice, the TimezoneFinder class was super slow

Reply:
Yeah and that timezonefinder package is huge. This isn't as accurate, but the speed up and smaller package size are worth it imo